evals/elsuite/audio/eval.py

```diff
@@ -0,0 +1,301 @@
+            format_kwargs={**completions, **test_sample, **self.modelgraded_spec_args},
+        )
+        evals.record.record_metrics(choice=choice, score=info["score"])
+
+    def run(self, recorder: RecorderBase):
+        super().run(recorder)
+        record_metrics = {}
+        events = recorder.get_metrics()
+        choices = [m["choice"] for m in events]
+        counts = dict(Counter(choices))
+        record_metrics.update({f"counts/{k}": v for k, v in counts.items()})
+        scores = [m["score"] for m in events if m["score"] is not None]
+        if scores:
+            record_metrics["score"] = sum(scores) / len(scores)
+        return record_metrics
+
+
+class Transcribe(MatchAudioTask):
+    def load_sample(self, row):
+        return Sample(audio=row["audio"]["array"], transcript=row["text"], expected=row["text"])
+
+    def build_prompt(self, sample: Sample):
+        return "Repeat after me in English:"
+
+    def compute_metrics(self, sample: Sample, sampled):
+        expected = sample.expected
+        score = self._compute_wer(expected, sampled)
+        evals.record.record_metrics(wer=score)
+        match = score < 0.1
+        evals.record.record_match(match, expected=expected, sampled=sampled, wer=score)
+        return match
+
+    def compute_corpus_metrics(self, events, expected, sampled):
+        return {
+            "accuracy": evals.metrics.get_accuracy(events),
+            "wer": self._compute_wer(expected, sampled),
+        }
+
+    def _compute_wer(self, expected, sampled):
+        transform = jiwer.Compose(
+            [
+                jiwer.RemovePunctuation(),
+                jiwer.ToLowerCase(),
+                jiwer.RemoveWhiteSpace(replace_by_space=True),
+                jiwer.Strip(),
+                jiwer.ReduceToListOfListOfWords(),
+            ]
+        )
+        output = jiwer.process_words(
+            expected, sampled, reference_transform=transform, hypothesis_transform=transform
+        )
+        return output.wer
+
+
+class Translate(MatchAudioTask):
+    def __init__(
+        self,
+        completion_fns: list[CompletionFn],
+        dataset: str,
+        target_language: str,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(completion_fns, dataset, *args, **kwargs)
+        self.target_language = target_language
+        self.bleu = BLEU(effective_order=True)
+
+    def load_sample(self, row):
+        return Sample(
+            audio=row["audio"]["array"], transcript=row["sentence"], expected=row["translation"]
+        )
+
+    def build_prompt(self, sample: Sample):
+        return "Translate the following into %s, without any explanation:" % self.target_language
+
+    def compute_metrics(self, sample: Sample, sampled: str):
+        expected = sample.expected
+        score = self.bleu.sentence_score(sampled, [expected]).score
+        evals.record.record_metrics(sacrebleu_sentence_score=score)
```

Comment:
`record_metrics` knows to compute average for "sample-based" metrics by default?